packages/ui/src/components/vuestic-components/va-image/VaImage.vue

```diff
@@ -26,90 +26,86 @@
   </div>
 </template>
 
-<script>
+<script lang="ts">
 import { makeContextablePropsMixin } from '../../context-test/context-provide/ContextPlugin'
+import { Mixins, Component, Prop, Watch } from 'vue-property-decorator'
 
-export default {
-  name: 'VaImage',
-  mixins: [
-    makeContextablePropsMixin({
-      ratio: {
-        type: [Number, String],
-        default: 1,
-      },
-      contain: {
-        type: Boolean,
-        default: false,
-      },
-    }),
-  ],
-  props: {
-    src: {
-      type: String,
-      required: true,
-    },
+@Component
+export default class VaImage extends Mixins(makeContextablePropsMixin({
+  ratio: {
+    type: [Number, String],
+    default: 1,
   },
+  contain: {
+    type: Boolean,
+    default: false,
+  },
+})) {
+  @Prop({
+    type: String,
+    required: true,
+  }) readonly src!: string
+
   data () {
     return {
       image: null,
-
       loading: false,
       loadingError: false,
     }
-  },
+  }
+
   beforeDestroy () {
     this.destroyLoader()
-  },
-  watch: {
-    src: {
-      handler () {
-        this.createLoader()
-      },
-      immediate: true,
-    },
-  },
-  computed: {
-    imageStyles () {
-      return {
-        'background-image': `url(${this.src})`,
-        'background-size': this.contain ? 'contain' : 'cover',
-      }
-    },
-    paddingStyles () {
-      return {
-        'padding-bottom': `${1 / this.c_ratio * 100}%`,
-      }
-    },
-  },
-  methods: {
-    createLoader () {
-      this.destroyLoader()
-      this.loading = true
-      this.loadingError = false
-      this.image = new Image()
-      this.image.onload = this.handleLoad
-      this.image.onerror = this.handleError
-      this.image.src = this.src
-    },
-    destroyLoader () {
-      if (this.image) {
-        this.image.onload = null
-        this.image.onerror = null
-        this.image = null
-      }
-    },
-    handleLoad () {
-      this.destroyLoader()
-      this.loading = false
-      this.$emit('loaded', this.src)
-    },
-    handleError (err) {
-      this.destroyLoader()
-      this.loadingError = true
-      this.loading = false
-      this.$emit('error', err)
-    },
-  },
+  }
+
+  @Watch('src')
+  onSrcChange () {
```

Comment:
where is immediate?

Reply:
Seems like it works correct without immediate, but have memory leak with it set to true.